Taarafo.Portal.Web/Services/Foundations/Comments/CommentService.Expections.cs

```diff
@@ -0,0 +1,154 @@
+using Taarafo.Portal.Web.Models.Comments.Exceptions;
+using Xeptions;
+
+namespace Taarafo.Portal.Web.Services.Foundations.Comments
+{
+    public partial class CommentService
+    {
+        private delegate ValueTask<Comment> ReturningCommentFunction();
+
+        private async ValueTask<Comment> TryCatch(ReturningCommentFunction returningCommentFunction)
+        {
+            try
+            {
+                return await returningCommentFunction();
+            }
+            catch (NullCommentException nullCommentException)
+            {
+                throw CreateAndLogValidationException(nullCommentException);
+            }
+            catch (InvalidCommentException invalidCommentException)
+            {
+                throw CreateAndLogValidationException(invalidCommentException);
+            }
+            catch (HttpRequestException httpRequestException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpRequestException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
+            catch (HttpResponseUrlNotFoundException httpResponseUrlNotFoundException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpResponseUrlNotFoundException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
+            catch (HttpResponseUnauthorizedException httpResponseUnauthorizedException)
+            {
+                var failedCommentDependencyException =
+                    new FailedCommentDependencyException(httpResponseUnauthorizedException);
+
+                throw CreateAndLogCriticalDependencyException(failedCommentDependencyException);
+            }
+            catch (HttpResponseNotFoundException httpResponseNotFoundException)
+            {
+                var notFoundCommentException =
+                    new NotFoundCommentException(httpResponseNotFoundException);
+
+                throw CreateAndLogDependencyValidationException(notFoundCommentException);
+            }
+            catch (HttpResponseBadRequestException httpResponseBadRequestException)
+            {
+                var invalidCommentException =
+                    new InvalidCommentException(
+                        httpResponseBadRequestException,
+                        httpResponseBadRequestException.Data);
+
+                throw CreateAndLogDependencyValidationException(invalidCommentException);
+            }
+            catch (HttpResponseConflictException httpResponseConflictException)
+            {
+                var invalidCommentException =
+                    new InvalidCommentException(
+                        httpResponseConflictException,
+                        httpResponseConflictException.Data);
```

Comment:
Do we need to transcend the data on conflict? What's in the dictionary?
I've been thinking about this for a bit. I think we might just need to keep it at the message level for now. what do you think?

Reply:
@hassanhabib Data is currently empty as we don't add anything on the API side when an AlreadyExistsException OR LockedException is thrown.   

I think it could be OK for message level for now OR we can keep Data in there as a standard for client side validation exception handling as this could be beneficial for the sake of the logger to get that captured automictically on the Portal in the event that data might be present, even though we know this is not specifically implemented in the API at the moment and not necessarily needed for UI validation or toast messages.   Happy either way, so just let me know your preference.